src/model/CoilHeatingElectricMultiStage_Impl.hpp

```diff
@@ -0,0 +1,114 @@
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_COILHEATINGELECTRICMULTISTAGE_IMPL_HPP
+#define MODEL_COILHEATINGELECTRICMULTISTAGE_IMPL_HPP
+
+#include "ModelAPI.hpp"
+#include "StraightComponent_Impl.hpp"
+
+namespace openstudio {
+namespace model {
+
+  class Schedule;
+  class CoilHeatingElectricMultiStageStageData;
+
+  namespace detail {
+
+    /** CoilHeatingElectricMultiStage_Impl is a StraightComponent_Impl that is the implementation class for CoilHeatingElectricMultiStage.*/
+    class MODEL_API CoilHeatingElectricMultiStage_Impl : public StraightComponent_Impl
+    {
+     public:
+      /** @name Constructors and Destructors */
+      //@{
+
+      CoilHeatingElectricMultiStage_Impl(const IdfObject& idfObject, Model_Impl* model, bool keepHandle);
+
+      CoilHeatingElectricMultiStage_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model, bool keepHandle);
+
+      CoilHeatingElectricMultiStage_Impl(const CoilHeatingElectricMultiStage_Impl& other, Model_Impl* model, bool keepHandle);
+
+      virtual ~CoilHeatingElectricMultiStage_Impl() {}
+
+      //@}
+      /** @name Virtual Methods */
+      //@{
+
+      virtual const std::vector<std::string>& outputVariableNames() const override;
+
+      virtual IddObjectType iddObjectType() const override;
+
+      virtual std::vector<ScheduleTypeKey> getScheduleTypeKeys(const Schedule& schedule) const override;
+
+      //@}
+      /** @name Getters */
+      //@{
+
+      boost::optional<Schedule> availabilitySchedule() const;
+
+      //@}
+      /** @name Setters */
+      //@{
+
+      bool setAvailabilitySchedule(Schedule& schedule);
+
+      void resetAvailabilitySchedule();
+
+      //@}
+      /** @name Other */
+      //@{
+
+      virtual unsigned inletPort() const override;
+
+      virtual unsigned outletPort() const override;
+
+      ModelObject clone(Model model) const override;
+
+      std::vector<ModelObject> children() const override;
```

Comment:
Should be in virtual section.

Reply:
Switching to ParentObject...

Reply:
Wait what? The StageData isn't a ParentObject, it doesn't have children.

Reply:
Unlike some other stagedata that do have curve. Here it is NOT a ParentObject

Reply:
What's the risk, though? Can't `children` just return empty?

Reply:
what's the point though? the whole point of ParentObject is to try (perhaps too cleverly) to handle the parent / child relationship automatically (eg: clone will call clonedChild.setParent(thisClone)). It also does matter a lot of your object has ResourceObject (which it doesn't). Keep it simple, ModelObject